mysqld_exporter.go

```diff
@@ -164,24 +238,76 @@ func (e *Exporter) scrape(scrapes chan<- []string) {
 	e.duration.Set(float64(time.Now().UnixNano()-now) / 1000000000)
 }
 
+func (e *Exporter) setGenericMetric(name string, value float64) {
+	if _, ok := e.metrics[name]; !ok {
+		e.metrics[name] = prometheus.NewUntyped(prometheus.UntypedOpts{
+			Namespace: namespace,
+			Name:      name,
+		})
+	}
+
+	e.metrics[name].Set(value)
+}
+
 func (e *Exporter) setMetrics(scrapes <-chan []string) {
-	for row := range scrapes {
+	var comRegex = regexp.MustCompile(`^com_(.*)$`)
+	var connectionErrorRegex = regexp.MustCompile(`^connection_errors_(.*)$`)
+	var handlerRegex = regexp.MustCompile(`^handler_(.*)$`)
+	var innodbPageRegex = regexp.MustCompile(`^innodb_pages_(.*)$`)
+	var innodbRowRegex = regexp.MustCompile(`^innodb_rows_(.*)$`)
+	var performanceSchemaRegex = regexp.MustCompile(`^performance_schema_(.*)$`)
+	var sslConnectionRegex = regexp.MustCompile(`^ssl_(.*_renegotiates|accepts|client_connects|connect_renegotiates|finished_.*|)$`)
+	var sslSessionCacheRegex = regexp.MustCompile(`^ssl_session_cache_(hits|misses|overflows|timeouts)$`)
```

Comment:
How do overflows and timeouts fit with hits and misses?

This one could require ongoing maintenance as new states appear